src/translations/maps/queuedPhysicalIndexToValueMap.js

```diff
@@ -0,0 +1,103 @@
+import IndexMap from './indexMap';
+import { getListWithRemovedItems, getListWithInsertedItems } from './utils/physicallyIndexed';
+import { getDecreasedIndexes, getIncreasedIndexes } from './utils/actionsOnIndexes';
+import { isFunction } from '../../helpers/function';
+
+/**
+ * Map for storing mappings from an physical index to a value. Some values are stored in a certain order.
+ *
+ * Does not update stored values on remove/add row or column action. Otherwise, queue of indexes related to ordered
+ * values is updated after such changes.
+ */
+class QueuedPhysicalIndexToValueMap extends IndexMap {
+  constructor() {
+    super();
+    this.queueOfIndexes = [];
+  }
+
+  /**
+   * Add values to list and reorganize. It updates queue of indexes related to ordered values.
+   *
+   * @private
+   * @param {number} insertionIndex Position inside the list.
+   * @param {Array} insertedIndexes List of inserted indexes.
+   */
+  insert(insertionIndex, insertedIndexes) {
+    this.indexedValues = getListWithInsertedItems(
+      this.indexedValues,
+      insertionIndex,
+      insertedIndexes,
+      this.initValueOrFn
+    );
+    this.queueOfIndexes = getIncreasedIndexes(this.queueOfIndexes, insertedIndexes);
+
+    super.insert(insertionIndex, insertedIndexes);
+  }
+
+  /**
+   * Remove values from the list and reorganize. It updates queue of indexes related to ordered values.
+   *
+   * @private
+   * @param {Array} removedIndexes List of removed indexes.
+   */
+  remove(removedIndexes) {
+    this.indexedValues = getListWithRemovedItems(this.indexedValues, removedIndexes);
+    this.queueOfIndexes = getDecreasedIndexes(this.queueOfIndexes, removedIndexes);
```

Comment:
In a case when you have for example sorted the second, the fifth, and six columns. The initial indexes are correctly stored as `[1, 4, 5]`. However, when we removing the columns where the sorting was applied (`hot.alter('remove_col', 0, 10)`) the `this.queueOfIndexes` looks like `[0, 0, 0]`. This causes weird results from the API for example `hot.getPlugin('multiColumnSorting').getSortConfig()` returns the three objects with columns `0` and adding a new column while pressing the <kbd>Ctrl/Cmd</kbd> key doesn't work anymore.

Reply:
Thanks. It should be fixed within https://github.com/handsontable/handsontable/pull/7276/commits/3ac658d899570aa90829e5f048d9ce038808ed83.